### PR TITLE
docs(csv_logs): Clarify CSV and YAML logging distinction and improve examples

### LINKED ISSUE #20724: CSVLogger missing information about yaml
### 📚 Documentation

The [CSVLogger](https://lightning.ai/docs/pytorch/stable/extensions/generated/lightning.pytorch.loggers.CSVLogger.html) claims to "Log to local file system in **yaml and CSV** format" but missing any usually information on how to configure it for yaml?

### PR DESCRIPTION
**Pull request by bhimrazy**
## What does this PR do?
Fixes #20724 

Improves the CSVLogger documentation to clearly explain the distinction between CSV and YAML logging formats and provides comprehensive examples.

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--20951.org.readthedocs.build/en/20951/

### CODE REVIEW

**Review by Borda — APPROVED** [review]

**Review by Borda — COMMENTED** [review]
this is great extension, @williamFalcon mind approve it? 🙏

**Review by lantiga — APPROVED** [review]
Looks great

File: src/lightning/pytorch/loggers/csv_logs.py [hunk 1]
```diff
@@ -55,22 +54,77 @@ def __init__(self, log_dir: str) -> None:
...
+
+        """
         self.hparams.update(params)
         hparams_file = os.path.join(self.log_dir, self.NAME_HPARAMS_FILE)
         save_hparams_to_yaml(hparams_file, self.hparams)
 
 
 class CSVLogger(Logger, FabricCSVLogger):
-    r"""Log to local file system in yaml and CSV format.
+    r"""Log to local file system in CSV and YAML format.
```
Comment (SkafteNicki) [inline src/lightning/pytorch/loggers/csv_logs.py:72]: could we add the same docstring to the fabric CSVLogger
https://github.com/Lightning-AI/pytorch-lightning/blob/master/src/lightning/fabric/loggers/csv_logs.py
just copy paste

File: src/lightning/pytorch/loggers/csv_logs.py [hunk 2]
```diff
@@ -140,6 +194,21 @@ def save_dir(self) -> str:
     @override
     @rank_zero_only
     def log_hyperparams(self, params: Optional[Union[dict[str, Any], Namespace]] = None) -> None:
+        """Log hyperparameters to YAML format.
```
Comment (SkafteNicki) [inline src/lightning/pytorch/loggers/csv_logs.py:197]: could we add the same docstring to the fabric CSVLogger
https://github.com/Lightning-AI/pytorch-lightning/blob/master/src/lightning/fabric/loggers/csv_logs.py
just copy paste

File: src/lightning/pytorch/loggers/csv_logs.py (hunk 1 above)
Comment (bhimrazy) [reply src/lightning/pytorch/loggers/csv_logs.py:72]: Thanks @SkafteNicki  for pointing this out. 

I had a look, but since the Fabric `CSVLogger` doesn’t handle hyperparameters (unlike the PyTorch `CSVLogger`), the docstring here wouldn’t really apply 1:1. I feel the current docstring in Fabric is already accurate for its scope.

But let me know wdyt — I can create a follow-up PR for the Fabric logger if needed.

**Review by SkafteNicki — APPROVED** [review]

**Comment by Borda** [comment]
@bhimrazy, mind checking the failing tests? 🦩

**Review by tchaton — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
